### ISSUE TITLE
ADIOS 2.5.0: core/Stream fails on i686

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS2#2137]
**Describe the bug**
ADIOS 2.5.0 does not build on Linux `i686` (docker image: `quay.io/pypa/manylinux2010_i686`) with GCC 7.3.1.

The build error is:
```
In file included from /project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:12:0:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = std::basic_string<char>; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
         Write(name, &datumLocal, {adios2::LocalValueDim}, {}, {}, vParams(),
         ^~~~~
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = signed char; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = short int; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = int; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = long long int; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = unsigned char; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = short unsigned int; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = unsigned int; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = long long unsigned int; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = float; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = double; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = long double; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = std::complex<float>; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc: In instantiation of ‘void adios2::core::Stream::Write(const string&, const T&, bool, bool) [with T = std::complex<double>; std::string = std::basic_string<char>]’:
/project/ADIOS2-2.5.0/source/adios2/core/Stream.cpp:196:1:   required from here
/project/ADIOS2-2.5.0/source/adios2/core/Stream.tcc:128:9: error: narrowing conversion of ‘18446744073709551613’ from ‘long long unsigned int’ to ‘unsigned int’ inside { } [-Wnarrowing]
make[2]: *** [source/adios2/CMakeFiles/adios2.dir/core/Stream.cpp.o] Error 1
```
**To Reproduce**

Travis CI job: https://travis-ci.com/github/openPMD/openPMD-api/jobs/319416157

PR: https://github.com/openPMD/openPMD-api/pull/717

**Expected behavior**
Compile :)

Maybe add this image/architecture to CI?

**Desktop (please complete the following information):**
 - OS/Platform: Ubuntu 18.04 on Travis-CI's instances within a [`quay.io/pypa/manylinux2010_i686` docker image](https://github.com/pypa/manylinux)
 - Build: gcc 7.3.1, cmake 3.17.1, build type: static

**Additional context**
Explored for binary wheel releases of [openPMD-api](https://github.com/openPMD/openPMD-api) for emerging and small-scale developer/IOT architectures.

**Comment by ax3l** [comment]
Could potentially be a problem with C++14 build, because I can get it to compile when switching explicitly to C++17: https://cloud.drone.io/conda-forge/adios2-feedstock/19/1/2

**Comment by pnorbert** [comment]
@ax3l @williamfgc Is this problem fixed with #2138? Can we close this issue?

**Comment by ax3l** [comment]
Yes, that fixed it according to my tests. Thanks again, William!
